File: backend/byteBlog/src/main/java/com/jeswin8801/byteBlog/entities/dto/blog/AuthorCompactDto.java
package com.jeswin8801.byteBlog.entities.dto.blog;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthorCompactDto {

    @JsonProperty("full-name")
    private String fullName;

    private String username;

    @JsonProperty("profile-image-url")
    private String profileImageUrl;

}

File: frontend/src/app/models/dtos/blog/author-compact-dto.ts
import { JsonProperty } from 'json-object-mapper';

export class AuthorCompactDto {
  @JsonProperty({ name: 'full-name' })
  fullName?: string;

  username?: string;

  @JsonProperty({ name: 'profile-image-url' })
  profileImageUrl?: string;
}
